app/controllers/withdraw_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class WithdrawController < ApplicationController
+  def create
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    edition = @document.current_edition
+    @public_explanation =
+      edition.withdrawn? ? edition.status&.details&.public_explanation : nil
```

Comment:
Minor - but I think we can remove the safe navigation operators here as we should expect a withdrawn document to have a details and public explanation and seems reasonable to throw exception if not